src/webgpu/util/texture/texel_data.spec.ts

```diff
@@ -106,16 +107,67 @@ function doTest(
   pass.end();
   t.device.queue.submit([encoder.finish()]);
 
-  t.expectGPUBufferValuesEqual(
-    outputBuffer,
-    new ReadbackTypedArray(
-      rep.componentOrder.map(c => {
-        const value = rep.decode(componentData)[c];
-        assert(value !== undefined);
-        return value;
-      })
-    )
+  const idealReadbackData = new ReadbackTypedArray(
+    rep.componentOrder.map(c => {
+      const value = rep.decode(componentData)[c];
+      assert(value !== undefined);
+      return value;
+    })
   );
+
+  if (format === 'rgba8unorm-srgb' || format === 'bgra8unorm-srgb') {
+    // In D3D the SRGB -> Float conversion is permitted a tolerance of 0.5f ULP on the SRGB side.
+    // The procedure for measuring this tolerance is to convert the result back into SRGB space
+    // using the ideal FLOAT -> SRGB conversion specified below but WITHOUT the rounding to integer,
+    // and taking the floating point difference versus the original SRGB value to yield the error.
+    // Exact conversion is required: 0.0f and 1.0f (the ends) must be exactly achievable.
+    // https://microsoft.github.io/DirectX-Specs/d3d/archive/D3D11_3_FunctionalSpec.htm#3.2.3.7%20SRGB%20-%3E%20FLOAT
```

Comment:
NIce fix! We've had problems here from the very beginning - a lot of tests that use the various texel data checking functions allow variances &le; 1 ULP, where really I wanted to allow variances &lt; 1 ULP (or for cases like this one, &le; 0.50001 ULP for "correctly rounded" results with a little tolerance for JS-side error) - but it seemed like a pretty difficult thing to fix so I decided it was OK to swallow the imprecision. Didn't realize this test had a real problem with it.